plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hadoop/TrinoHadoopCatalog.java

```diff
@@ -0,0 +1,317 @@
+    {
+        return EMPTY_SESSION_MAP;
+    }
+
+    private Catalog createNewCatalog(ConnectorSession session)
+    {
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        builder.putAll(catalogProperties);
+        builder.putAll(getSessionProperties(session));
+        Configuration hadoopConf = hdfsEnvironment.getConfiguration(new HdfsEnvironment.HdfsContext(session), new Path(warehouse));
+        return loadCatalog(CATALOG_IMPL, catalogName.toString(), builder.buildOrThrow(), hadoopConf);
+    }
+
+    public Catalog getCatalog(ConnectorSession session)
+    {
+        try {
+            return catalogCache.get(getSessionCacheKey(session), () -> createNewCatalog(session));
+        }
+        catch (ExecutionException e) {
+            throw new IllegalStateException("Fail to create catalog for " + session, e);
+        }
+    }
+
+    private SupportsNamespaces getNamespaceCatalog(ConnectorSession session)
+    {
+        Catalog catalog = getCatalog(session);
+        if (catalog instanceof SupportsNamespaces) {
+            return (SupportsNamespaces) catalog;
+        }
+        throw new TrinoException(NOT_SUPPORTED, "catalog " + CATALOG_IMPL + " does not support namespace operations");
+    }
+
+    private Map<String, String> convertToCatalogProperties(IcebergConfig config)
+    {
+        return ImmutableMap.of(WAREHOUSE_LOCATION, config.getCatalogWarehouse());
+    }
+
+    private TableIdentifier toTableId(SchemaTableName schemaTableName)
+    {
+        return TableIdentifier.of(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+    }
+
+    private SchemaTableName schemaFromTableId(TableIdentifier tableIdentifier)
+    {
+        return new SchemaTableName(tableIdentifier.namespace().toString(), tableIdentifier.name());
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return getNamespaceCatalog(session).listNamespaces().stream().map((namespace) -> {
+            return namespace.toString();
+        }).collect(Collectors.toList());
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        SupportsNamespaces catalog = getNamespaceCatalog(session);
+        catalog.dropNamespace(catalog.listNamespaces().stream().filter(_namespace -> _namespace.toString() == namespace).findAny().orElse(null));
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return (Map) getNamespaceCatalog(session).loadNamespaceMetadata(Namespace.of(namespace));
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported by " + getCatalog(session).name());
```

Comment:
Copy / paste error here
```suggestion
        throw new TrinoException(NOT_SUPPORTED, "getNamespacePrincipal is not supported by " + getCatalog(session).name());
```